src/main/java/com/mg105/interface_adapters/BattlePresenter.java

```diff
@@ -1,72 +1,39 @@
 package com.mg105.interface_adapters;
 
-import com.mg105.entities.Battle;
-import com.mg105.entities.BattleCharacter;
-import com.mg105.entities.GameState;
-import com.mg105.entities.Move;
-import com.mg105.user_interface.BattleMenu;
+import com.mg105.use_cases.BattlePresenterInterface;
+import com.mg105.use_cases.BattleInteractor;
 
 import java.util.ArrayList;
-import java.util.Random;
 
 /**
- * Class communicating between BattleMenu view and game entities.
- *
- * Follows MVP pattern; contains a reference to the view and interacts with Battle and related classes.
+ * Class communicating between BattleMenuInterface view and BattleInteractor.
+ * Follows MVP pattern; contains a reference to the view and interacts with BattleInteractor.
  */
-public class BattlePresenter {
+public class BattlePresenter implements BattlePresenterInterface {
 
-    private GameState state;
-    private Battle encounter;
+    private final BattleInteractor interactor;
 
-    private BattleMenu view;
+    private BattleMenuInterface view;
 
-    public BattlePresenter(GameState state) {
-        this.state = state;
-        this.encounter = state.getCurrEncounter();
-        BattleMenu.setPresenter(this);
+    public BattlePresenter(BattleInteractor interactor) {
+        this.interactor = interactor;
+        this.interactor.setPresenter(this);
     }
 
     /**
      * Sets the view attribute.
      *
-     * @param view the BattleMenu instance to set the view attribute to.
+     * @param view the BattleMenuInterface instance to set the view attribute to.
```

Comment:
Its better practice to put @Override here when implementing methods from an interface